### PR TITLE
Introduce feature flag constants

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Goals: 1) prevent errors in feature flag use, 3) improve readability of feature flags by enabling renaming constant used at call-sites, 3) make it easier to audit/retire feature flags
- Defines feature flag constants
- Drops integration banner flag from code now that's in enabled for everyone (and it's not a kill switch)
- Fixes bug in feature flag check for showing Page Editor instructions action in Quick Bar

## Future Work

- Do similar refactoring for milestones

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/components/quickBar/QuickBarApp.tsx [hunk 1]
```diff
@@ -223,7 +224,7 @@ export const initQuickBarApp = once(async () => {
     quickBarRegistry.addAction(action);
   }
 
-  if (!(await flagOn("page-editor"))) {
+  if (!(await restrict(RestrictedFeatures.PAGE_EDITOR))) {
```
Comment (twschiller) [inline src/components/quickBar/QuickBarApp.tsx:227]: 🐛  bug fix

File: src/auth/featureFlags.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+   */
+  REPORT_BACKGROUND_DEPLOYMENTS: "report-background-deployments",
+
+  /**
+   * Enable additional metadata with error telemetry for enterprise customers.
+   *
+   * Controlled by the backend, not waffle:
+   * https://github.com/pixiebrix/pixiebrix-app/blob/9ecd1d0357d9221e91dca96b3dfea16dcef177f4/api/serializers/account.py#L326-L326
+   */
+  // XXX: move to its own constant b/c it's not controlled by waffle?
```
Comment (twschiller) [inline src/auth/featureFlags.ts:37]: This is the odd one out, it's controlled by organization policy

File: src/auth/featureFlags.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+   */
+  PAGE_EDITOR_DYNAMIC_QUICKBAR: "pageeditor-quickbar-provider",
+} as const;
+
+/**
+ * @see FeatureFlags
+ * @see OrganizationFlags
+ */
+export type FeatureFlag =
+  // The constants are split out to make it easier to see which flags are controlled by the organization policy.
```
Comment (twschiller) [inline src/auth/featureFlags.ts:127]: For discussion - easy to refactor later

Comment (grahamlangford) [reply src/auth/featureFlags.ts:127]: I like it. This is another advantage of switching to constants

**Review by grahamlangford — APPROVED** [review]

File: src/components/quickBar/QuickBarApp.tsx (hunk 1 above)
Comment (BLoe) [reply src/components/quickBar/QuickBarApp.tsx:227]: What do you think about renaming the `restrict/permit` functions to be a bit more clear what's going on? Also, do you think we definitely need both of them? What if there was just one function called `hasRestrictedFlagForFeature()` or something along those lines? The action-verbs-used-as-booleans (restrict/permit) is just still kind of unintuitive to me.

**Review by BLoe — APPROVED** [review]

**Comment by twschiller** [comment]
> What do you think about renaming the restrict/permit functions to be a bit more clear what's going on? Also, do you think we definitely need both of them? 

They have different semantics for what happens if the flags are still fetching. Will leave as-is for now in this PR

**Pull request merged**